### PR TITLE
Validate Docker credentials

### LINKED ISSUE #3625: [builder ui] Validate Docker credentials
As a user, when I enter my docker hub creds on the integrations tab
I want to see visual confirmation that they were correct/validated
so that my build doesn't fail silently
requiring me to spend my time figuring out this reason.

### To-do

- [x]  Create mockup; add to this issue
- [x]  Add validation in backend/UI

### Mockup
[Mockup for validation error on creds](https://chef.invisionapp.com/share/5YDV6902R#/251489938_my_Origins_Integrations)

### PR DESCRIPTION
**Pull request by cnunciato**
This change adds a new API endpoint and associated UI support for validating Docker credentials. Also adds a mechanism for unsubscribing to updates on the Redux store.

Signed-off-by: Christian Nunciato <cnunciato@chef.io>

Fixes #3625 and #3677.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-api/src/server/handlers.rs [hunk 1]
```diff
@@ -119,6 +125,60 @@ pub fn job_group_promote(req: &mut Request) -> IronResult<Response> {
...
+            return Ok(Response::with(status::BadRequest));
+        }
+    };
+
+    if !body["username"].is_string() || !body["password"].is_string() {
+        debug!("Error: Missing username or password");
+        return Ok(Response::with(status::BadRequest));
+    }
+
+    // There's probably a better way to do this. Suggestions?
```
Comment (raskchanky) [inline components/builder-api/src/server/handlers.rs]: The only other thing that occurs to me is shelling out to `docker` and running e.g. `docker login` and checking its exit code.  One nice thing about that is if they change their API in the future, we don't have to change our API call.  Honestly though, it's probably not a big deal.  This looks fine too.

Comment (cnunciato) [reply components/builder-api/src/server/handlers.rs]: I actually did look into that! But it wasn't clear `docker` would be installed on the API nodes, and it didn't seem necessary to install it, etc., just for this use case.

Comment (raskchanky) [reply components/builder-api/src/server/handlers.rs]: Yeah, good points

**Pull request merged**